In [45]:
import pandas as pd
import numpy as np
import matplotlib as plt
pd.set_option('display.max_columns', None)
pd.set_option("max_colwidth", 100)

Necessário excluir as 2 últimas linhas do arquivo para evitar 2 linhas nulas. das bases de 2009 até 2011.
A base de 2018 possui algumas colunas 'nome fantasia' que possuem ';' como separador de empresas e que
consequentemente atrapalham na leitura dos dados pois alteram a quantidade de colunas finais.
Sendo necessário a modificação por linha manual (18 linhas no total)

In [46]:
rec2009 = pd.read_csv("../data/reclamacoes-fundamentadas-sindec-2009.csv",header=0,encoding="ISO-8859-9",delimiter=";", dtype="str")
rec2010 = pd.read_csv("../data/reclamacoes-fundamentadas-sindec-2010.csv",header=0,encoding="ISO-8859-9",delimiter=";", dtype="str")
rec2011 = pd.read_csv("../data/reclamacoes-fundamentadas-sindec-2011.csv",header=0,encoding="ISO-8859-9",delimiter=";", dtype="str")
rec2012 = pd.read_csv("../data/reclamacoes-fundamentadas-sindec-2012.csv",header=0,encoding="UTF-8",delimiter=";", dtype="str")
rec2013 = pd.read_csv("../data/reclamacoes-fundamentadas-sindec-2013.csv",header=0,encoding="UTF-8",delimiter=";", dtype="str")
rec2014 = pd.read_csv("../data/reclamacoes-fundamentadas-sindec-2014.csv",header=0,encoding="UTF-8",delimiter=";", dtype="str")
rec2015 = pd.read_csv("../data/reclamacoes-fundamentadas-sindec-2015.csv",header=0,encoding="UTF-8",delimiter=";", dtype="str")
rec2016 = pd.read_csv("../data/reclamacoes-fundamentadas-sindec-2016_v2.csv",header=0,encoding="UTF-8",delimiter=";", dtype="str")
rec2017 = pd.read_csv("../data/CNRF_2017.csv",header=0,encoding="UTF-8",delimiter=";", dtype="str")
rec2018 = pd.read_csv("../data/CNRF2018.csv",header=0,encoding="UTF-8",delimiter=";", dtype="str")

In [47]:
rec2009 = rec2009.rename(columns={
    "anocalendario": "AnoCalendario",
    "codigoregiao": "CodigoRegiao",
    "regiao": "Regiao",
    "radicalCNPJ": "RadicalCNPJ"})


In [48]:
rec2010 = rec2010.rename(columns={
           "anocalendario": "AnoCalendario",
           "codigoregiao": "CodigoRegiao",
           "regiao": "Regiao",
           "radicalCNPJ": "RadicalCNPJ"})

In [49]:
rec2011 = rec2011.rename(columns={
           "anocalendario": "AnoCalendario",
           "codigoregiao": "CodigoRegiao",
           "regiao": "Regiao",
           "radicalCNPJ": "RadicalCNPJ"})

In [50]:
rec2012.columns

Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')

In [51]:
rec2013.columns

Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')

In [52]:
rec2014.columns

Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')

In [53]:
rec2015.columns

Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')

In [54]:
rec2016.columns

Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')

In [55]:
rec2017.columns

Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')

In [56]:
rec2018.columns

Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')

In [57]:
frames = [rec2009,rec2010,rec2011,rec2012,rec2013,rec2014,rec2015,rec2016,rec2017,rec2018]

In [58]:
df = pd.concat(frames)


A união de todas as tabelas deu 1.668.061 linhas com 23 colunas.


In [59]:
df

AnoCalendario         DataArquivamento             DataAbertura  \
0              2009  2009-01-21 15:29:29.000  2005-07-06 08:32:23.000   
1              2009  2009-05-12 12:05:08.000  2006-01-17 12:10:53.000   
2              2009  2009-04-23 08:52:31.000  2005-07-06 11:28:47.000   
3              2009  2009-04-23 08:52:31.000  2005-07-06 11:28:47.000   
4              2009  2009-07-23 11:57:40.000  2006-01-18 15:35:29.000   
...             ...                      ...                      ...   
39055          2018  2018-05-29 14:19:32.000  2017-01-30 14:46:11.000   
39056          2018  2018-06-06 13:38:04.000  2018-04-09 08:39:46.000   
39057          2018  2018-03-23 14:45:20.000  2018-02-23 16:37:15.000   
39058          2018  2018-04-09 11:41:30.000  2018-02-27 11:40:27.000   
39059          2018  2018-05-24 14:24:46.000  2018-03-07 13:17:07.000   

      CodigoRegiao        Regiao  UF  \
0                5  Centro-oeste  GO   
1                3       Sudeste  RJ   
2                5  Centro-oeste  GO   
3                5  Centro-oeste  GO   
4                3       Sudeste  RJ   
...            ...           ...  ..   
39055            3       Sudeste  MG   
39056            3       Sudeste  MG   
39057            2      Nordeste  RN   
39058            2      Nordeste  RN   
39059            2      Nordeste  RN   

                                           strRazaoSocial  \
0           CBP SUL - COLCHÕES E ESPUMAS INDUSTRIAIS LTDA   
1                                               GRADIENTE   
2                                      IGL INDÚSTRIA LTDA   
3                          UNILEVER BESTFOODS BRASIL LTDA   
4              EMPRESA BRASILEIRA DE TELECOMUNICAÇÕES S.A   
...                                                   ...   
39055                  BANCO LOSANGO S.A - BANCO MULTIPLO   
39056        MERCADOLIVRE.COM ATIVIDADES DE INTERNET LTDA   
39057                    ACBZ IMPORTACAO E COMERCIO LTDA.   
39058                     NET SERVIÇOS DE COMUNICAÇÃO S/A   
39059  CREFISA S.A. CREDITO FINANCIAMENTO E INVESTIMENTOS   

                   strNomeFantasia Tipo      NumeroCNPJ RadicalCNPJ  \
0                         LIMANSKY    1   1350934000116     1350934   
1                              NaN    1  43185362001936    43185362   
2      ELIDA PONDS INDUSTRIAL LTDA    1   3085759000102     3085759   
3                         UNILEVER    1   1615814002066     1615814   
4                 EMBRATEL - LIVRE    1  33530486000129    33530486   
...                            ...  ...             ...         ...   
39055                          NaN    1  33254319000100    33254319   
39056                MERCADO LIVRE    1   3361252000134     3361252   
39057               ASUS NO BRASIL    1   9509531000189     9509531   
39058                          NET    1    108786018889      108786   
39059                  CREFISA S/A    1             NaN         NaN   

                                            RazaoSocialRFB  \
0            CBP SUL - COLCHOES E ESPUMAS INDUSTRIAIS LTDA   
1                                       IGB ELETRONICA S.A   
2           UNILEVER BRASIL HIGIENE PESSOAL E LIMPEZA LTDA   
3                          UNILEVER BRASIL INDUSTRIAL LTDA   
4      EMPRESA BRASILEIRA DE TELECOMUNICACOES S A EMBRATEL   
...                                                    ...   
39055          HSBC FINANCE (BRASIL) S.A. - BANCO MULTIPLO   
39056         MERCADOLIVRE.COM ATIVIDADES DE INTERNET LTDA   
39057                     ACBZ IMPORTACAO E COMERCIO LTDA.   
39058                      NET SERVICOS DE COMUNICACAO S/A   
39059                                                  NaN   

                    NomeFantasiaRFB CNAEPrincipal  \
0                               NaN       3104700   
1                               NaN       2640000   
2      ELIDA PONDS INDUSTRIAL LTDA.       2063100   
3                               NaN       1031700   
4                               NaN       6110899   

In [60]:
def view_df_stats(df):
    print("Shape of df={}".format(df.shape))
    print("Number of index levels:{}".format(df.index.nlevels))
    for i in range(df.index.nlevels):
        print("For index level {},unique values count={}".format(i,df.index.get_level_values(i).unique().shape[0]))
    print("Columns of df={}".format(df.columns))
    print("Null count= \n {}".format(df.isnull().sum()))
    print(df.describe())
    

In [63]:
view_df_stats(df)


Shape of df=(1668061, 23)
Number of index levels:1
For index level 0,unique values count=268096
Columns of df=Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')
Null count= 
 AnoCalendario                 0
DataArquivamento             20
DataAbertura                  7
CodigoRegiao                  0
Regiao                        0
UF                            0
strRazaoSocial               11
strNomeFantasia          267963
Tipo                          0
NumeroCNPJ                71583
RadicalCNPJ               73166
RazaoSocialRFB            93932
NomeFantasiaRFB          912164
CNAEPrincipal

In [70]:
df[(df['CEPConsumidor'].isnull())].groupby("AnoCalendario").count()

DataArquivamento  DataAbertura  CodigoRegiao  Regiao     UF  \
AnoCalendario                                                                
2009                       9824          9824          9824    9824   9824   
2010                      11725         11725         11725   11725  11725   
2011                       9162          9162          9162    9162   9162   
2012                      13906         13906         13906   13906  13906   
2013                      18667         18667         18667   18667  18667   
2014                      17936         17936         17936   17936  17936   
2015                      18211         18211         18211   18211  18211   
2016                      14086         14086         14086   14086  14086   
2017                       4696          4696          4702    4702   4702   
2018                       1858          1858          1859    1859   1859   

               strRazaoSocial  strNomeFantasia   Tipo  NumeroCNPJ  \
AnoCalendario                                                       
2009                     9824             8025   9824        9304   
2010                    11725             9809  11725       11175   
2011                     9162             7964   9162        8824   
2012                    13906            11661  13906       13240   
2013                    18667            15131  18667       17996   
2014                    17936            15002  17936       17176   
2015                    18211            15148  18211       17153   
2016                    14086            11988  14086       13289   
2017                     4702             4057   4702        4364   
2018                     1859             1560   1859        1806   

               RadicalCNPJ  RazaoSocialRFB  NomeFantasiaRFB  CNAEPrincipal  \
AnoCalendario                                                                
2009                  9297            9285             4643           9285   
2010                 11157           11138             5696          11138   
2011                  8810            8798             4461           8798   
2012                 13162           13151             6323          13151   
2013                 17996           17897             8798          17552   
2014                 17137           17045             8851          17045   
2015                 17153           16809             8856          16809   
2016                 13289           12761             6688          12761   
2017                  4364            4093             2245           4093   
2018                  1797            1660              861           1660   

               DescCNAEPrincipal  Atendida  CodigoAssunto  DescricaoAssunto  \
AnoCalendario                                                                 
2009                        9269      9824           9824              9824   
2010                       10920     11725          11725             11725   
2011                        8614      9162           9162              9162   
2012                       12851     13906          13906             13906   
2013                       17552     18667          18667             18667   
2014                       16769     17936          17936             17936   
2015                       16614     18211          18211             18211   
2016                       12657     14086          14086             14086   
2017                        4062      4702           4702              4702   
2018                        1646      1859           1859              1859   

               CodigoProblema  DescricaoProblema  SexoConsumidor  \
AnoCalendario                                                      
2009                     9824               9824            9824   
2010                    11725              11725           11725   
2011                     9162               9162            9162   
2012                   

Verificando que o número de CEPs que são vazios é de 120.078 e que isso representa
 7,20% da base não é uma perda muito significativa que compromenta o estudo de caso.
 Dos 7,20%

 | Ano Calendario | Linhas |
 |----------------|--------|
 | 2009           | 9824   |
 | 2010           | 11725  |
 | 2011           | 9162   |
 | 2012           | 13906  |
 | 2013           | 18667  |
 | 2014           | 17936  |
 | 2015           | 18211  |
 | 2016           | 14086  |
 | 2017           | 4696   |
 | 2018           | 1858   |


In [71]:
df2 = df.dropna(subset=["CEPConsumidor"])

In [72]:
view_df_stats(df2)

Shape of df=(1547983, 23)
Number of index levels:1
For index level 0,unique values count=268029
Columns of df=Index(['AnoCalendario', 'DataArquivamento', 'DataAbertura', 'CodigoRegiao',
       'Regiao', 'UF', 'strRazaoSocial', 'strNomeFantasia', 'Tipo',
       'NumeroCNPJ', 'RadicalCNPJ', 'RazaoSocialRFB', 'NomeFantasiaRFB',
       'CNAEPrincipal', 'DescCNAEPrincipal', 'Atendida', 'CodigoAssunto',
       'DescricaoAssunto', 'CodigoProblema', 'DescricaoProblema',
       'SexoConsumidor', 'FaixaEtariaConsumidor', 'CEPConsumidor'],
      dtype='object')
Null count= 
 AnoCalendario                 0
DataArquivamento             13
DataAbertura                  0
CodigoRegiao                  0
Regiao                        0
UF                            0
strRazaoSocial               11
strNomeFantasia          248230
Tipo                          0
NumeroCNPJ                65832
RadicalCNPJ               67250
RazaoSocialRFB            86491
NomeFantasiaRFB          849508
CNAEPrincipal

Antes da remoção dos CEPs nulos, o ano que possuia mais reclamações
era o de 2013 com 268.096 registros,após a remoção passou a ser o
de 2014 o que possui mais reclamações com 249.828 registros

Plot Faixa etária por quantidade de reclamações.
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.plot.bar.html

In [ ]:
from collections import Counter
age_counter = Counter(df['FaixaEtariaConsumidor'])
df = pd.DataFrame.from_dict(age_counter, orient='index')
df.plot(kind='bar')
